Added table (#780)
digital-asset/docs.daml.com: docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

-  A new interface instance can be added to a template.

> The following table summarizes the changes that SCU supports.  Please consult
> the detailed documentation below for additional information.

Reviewer: ```suggestion
The following table summarizes the changes supported by that SCU. Consult
the detailed documentation below for additional information.
```
Add link to "below"?

+-----------------+---------------+------------+-----------------------------------+
| Rename of       | Template      | No         |                                   |
| Template        | Definition    |            |                                   |
+-----------------+---------------+------------+-----------------------------------+
| Change of       | Template      | No         |                                   |
| Field Type      | Definition    |            |                                   |
+-----------------+---------------+------------+-----------------------------------+
| Change of       | Template      | No         |                                   |
| Field           | Definition    |            |                                   |
| Ordering        |               |            |                                   |
+-----------------+---------------+------------+-----------------------------------+
| Change of       | Template      | No         |                                   |
| Field Name      | Definition    |            |                                   |
+-----------------+---------------+------------+-----------------------------------+

> | Change of       | Template      | No         | A new template version needs to   |

Reviewer 2: Unsure how I feel about marking this as simply "Not covered by SCU". We have special logic to allow these to change with new contracts, and to recompute previous values.
Perhaps "Covered by SCU" isn't the right phrasing here, but instead something akin to "Can change in an upgrade"

| Signatories     | Definition    |            | return the same signatory result  |
|                 |               |            | for a persisted contract,         |
|                 |               |            | otherwise the contracts cannot be |
|                 |               |            | used by an upgraded package (this |
|                 |               |            | fails at runtime).                |
|                 |               |            |                                   |
|                 |               |            | One can change the function that  |
|                 |               |            | computes the signatories (for     |
|                 |               |            | example to take into account new  |
|                 |               |            | fields) when creating contracts.  |
+-----------------+---------------+------------+-----------------------------------+

> | Change of       | Template      | No         | The key's type cannot be changed, |

Reviewer 2: Should this not be "Change Of Key"?

> | Change of       | Template      | No         | The type of a key cannot be       |

Reviewer: This is awkward and needs to be rephrased - perhaps something like "The key's type cannot be changed - for example, you cannot add a new field to a record that the key contains."

Author: Change

| Keys            | Definition    |            | changed, like adding a new field  |
|                 |               |            | in the record it contains.        |
|                 |               |            |                                   |
|                 |               |            | One can change the function that  |
|                 |               |            | computes the key in the same way  |
|                 |               |            | the function computing the        |
|                 |               |            | signatories can be changed. A     |
|                 |               |            | persisted contract can be used by |
|                 |               |            | an upgrade only if the result of  |
|                 |               |            | the function does not change.     |
+-----------------+---------------+------------+-----------------------------------+
| Change of       | Template      | No         | Same as template signatories      |
| Observers       | Definition    |            |                                   |
+-----------------+---------------+------------+-----------------------------------+

> | Change of key   | Template      | No         | Same as template signatories      |

Reviewer: ```suggestion
| Change of Key   | Template      | No         | Same as template signatories      |
```

> | maintainers     | Definition    |            |                                   |

Reviewer: ```suggestion
| Maintainers     | Definition    |            |                                   |
```

|                 |               |            |                                   |
+-----------------+---------------+------------+-----------------------------------+
| Change of       | Template      | Yes        | Note that the Ensure is           |
| Ensure          | Definition    |            | recalculated just before usage so |
| Predicate       |               |            | any changes need to ensure that a |
|                 |               |            | V1 contract is still valid as a   |
|                 |               |            | V2 contract.                      |
+-----------------+---------------+------------+-----------------------------------+

> | Addition of     | Template      | Yes, but   |                                   |
> | new Definition  | Definition    | they they  |                                   |
> | fields, enum,   |               | must be    |                                   |
> | and variant     |               | optional   |                                   |
> | constructors    |               |            |                                   |

Reviewer: ```suggestion
| Addition of     | Template      | Yes, but   |                                   |
| New Definition  | Definition    | they they  |                                   |
| Fields, Enum,   |               | must be    |                                   |
| and Variant     |               | optional   |                                   |
| Constructors    |               |            |                                   |
```

+-----------------+---------------+------------+-----------------------------------+

> | Deletion of     | Template      | No         |                                   |
> | fields from     | Definition    |            |                                   |
> | template        |               |            |                                   |

Reviewer: ```suggestion
| Deletion of     | Template      | No         |                                   |
| Fields from     | Definition    |            |                                   |
| Template        |               |            |                                   |
```

| Change in       | Choice        | Yes        |                                   |
| Choice          |               |            |                                   |
| Controller      |               |            |                                   |
+-----------------+---------------+------------+-----------------------------------+
| Addition of     | Choice        | Yes, but   |                                   |
| Choice          |               | they must  |                                   |
| Parameters      |               | be         |                                   |
|                 |               | optional   |                                   |
|                 |               |            |                                   |
+-----------------+---------------+------------+-----------------------------------+
| Deletion of     | Choice        | No         |                                   |
| Choice          |               |            |                                   |
| Parameters      |               |            |                                   |
+-----------------+---------------+------------+-----------------------------------+

> | Change of       | Choice        | No         |                                   |
> | Consuming       |               |            |                                   |
> | choice to       |               |            |                                   |
> | Non-Consuming   |               |            |                                   |
> | Choice, or      |               |            |                                   |
> | vice-versa      |               |            |                                   |

Reviewer: ```suggestion
| Change of       | Choice        | No         |                                   |
| Consuming       |               |            |                                   |
| Choice to       |               |            |                                   |
| Non-Consuming   |               |            |                                   |
| Choice, or      |               |            |                                   |
| Vice-versa      |               |            |                                   |
```

+-----------------+---------------+------------+-----------------------------------+
| Change of       | Choice        | Yes        |                                   |
| Choice Body     |               |            |                                   |
+-----------------+---------------+------------+-----------------------------------+

> | Change of       | Choice        | Yes,       | You can change a user data type   |
> | Choice return   |               | except for | (record, variant, enum) contained |
> | type            |               | tuple or   | in the return type. In this case  |

Reviewer: ```suggestion
| Change of       | Choice        | Yes,       | You can change a user data type   |
| Choice Return   |               | except for | (record, variant, enum) contained |
| Type            |               | tuple or   | in the return type. In this case  |
```

|                 |               | scalar     | you can add a field to the record |
|                 |               | types      | or a new case to the              |
|                 |               |            | variant/enum.                     |
|                 |               |            |                                   |
|                 |               |            | In particular: you cannot change  |
|                 |               |            | a collection, Optional, List, or  |
|                 |               |            | Map; you can change Tuple, (to a  |
|                 |               |            | triple) if you define your own    |
|                 |               |            | type.                             |

> |                 |               |            |                                   |

Reviewer: Make capitalization consistent in this list

Author: Changed
